src/backend/types.ml

```diff
@@ -0,0 +1,125 @@
+
+(* This file is free software, part of FTW. See file "LICENSE" for more information *)
+
+module S = Openapi_router.Json_Schema
+
+(* Helper functions *)
+(* ************************************************************************* *)
+
+let obj o = S.Obj o
+let ref name = S.Ref name
+
+let int = obj S.Integer
+let array = obj S.Array
+let string = obj S.String
+let object_ = obj S.Object
+
+let schemas, make_schema =
+  let l = Stdlib.ref [] in
+  let schemas router =
+    List.fold_left (fun router (name, schema) ->
+        Router.schema name (obj schema) router
+      ) router !l
+  in
+  let make_schema ~name
+      ?schema ?id_ ?title
+      ?description ?default ?examples
+      ?read_only ?write_only ?comment
+      ?typ ?enum ?const ?multiple_of
+      ?minimum ?exclusiveMinimum ?maximum ?exclusiveMaximum
+      ?properties ?pattern_properties ?additional_properties ?required
+      ?property_names ?min_properties ?max_properties
+      ?min_items ?max_items ?items ?prefix_items
+      ?format ?pattern ?min_length ?max_length
+      ?all_of ?any_of ?one_of
+      () =
+    let schema =
+      S.make_schema ?schema ?id_ ?title
+        ?description ?default ?examples
+        ?read_only ?write_only ?comment
+        ?typ ?enum ?const ?multiple_of
+        ?minimum ?exclusiveMinimum ?maximum ?exclusiveMaximum
+        ?properties ?pattern_properties ?additional_properties ?required
+        ?property_names ?min_properties ?max_properties
+        ?min_items ?max_items ?items ?prefix_items
+        ?format ?pattern ?min_length ?max_length
+        ?all_of ?any_of ?one_of
+        ()
+    in
+    let res = name, schema in
+    l := res :: !l;
+    res
+  in
+  schemas, make_schema
+
+
+(* Schemas & conversions definitions *)
+(* ************************************************************************* *)
+
+(* Errors *)
+module Error = struct
+
+  type t = Error.t = {
+    message : string;
+  } [@@deriving yojson]
+
+  let ref, schema =
+    make_schema ()
+      ~name:"Error"
+      ~typ:object_
+      ~properties:[
+        "message", obj @@ S.make_schema ()
+          ~typ:string
+          ~examples:[`String "Event not found"]
+      ]
+
+
+end
+
+(* Event Ids *)
+module EventId = struct
+
+  type t = Ftw.Event.id [@@deriving yojson]
+
+  let ref, schema =
+    make_schema ()
+      ~name:"EventId"
+      ~typ:int
+      ~format:"int64"
+      ~examples:[`Int 42]
+end
+
+(* Event Id list *)
+module EventIdList = struct
+  type t = {
+    events : Ftw.Event.id list;
+  } [@@deriving yojson]
+
+  let ref, schema =
+    make_schema ()
+      ~name:"EventIdList"
+      ~typ:object_
+      ~properties:[
+        "events", obj @@ S.make_schema ()
+          ~typ:array
+          ~items:(ref EventId.ref);
+      ]
+end
+
+(* Event specification *)
+module Event = struct
+  type t = {
+    name : string;
```

Comment:
shouldn't you add start_date and end_date there as well ? to comply with api endpoints declared above

Reply:
done